internal/scan-manager/webapp/src/main/java/org/wso2/security/tools/scanmanager/webapp/filter/JWTValidationFilter.java

```diff
@@ -0,0 +1,129 @@
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import org.springframework.web.servlet.HandlerExceptionResolver;
+import org.wso2.security.tools.scanmanager.webapp.exception.ScanManagerWebappException;
+import org.wso2.security.tools.scanmanager.webapp.util.Constants;
+import org.wso2.security.tools.scanmanager.webapp.util.JwtTokenUtil;
+
+import java.io.IOException;
+import java.util.HashMap;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+
+/**
+ * Filter class to validate JWT token.
+ */
+@Component
+public class JWTValidationFilter implements Filter {
+
+    @Value("${security.key.store.name}")
+    private String keyStoreName;
+
+    @Value("${security.key.store.pwd}")
+    private String keyStorePwd;
+
+    @Value("${security.key.alias}")
+    private String alias;
+
+    @Value("${security.signing.algo}")
+    private String algorithm;
+
+    @Value("${jwt.issuer}")
+    private String jwtIssuer;
+
+    @Value("${jwt.audience}")
+    private String jwtAudience;
+
+    @Value("${jwt.expiry}")
+    private String jwtExpiry;
+
+    @Autowired
+    @Qualifier("handlerExceptionResolver")
+    private HandlerExceptionResolver resolver;
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+    }
+
+    @Override
+    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain filterChain)
+            throws IOException, ServletException {
+
+        String username;
+        if (servletRequest instanceof HttpServletRequest) {
+            HttpServletRequest request = (HttpServletRequest) servletRequest;
+            HttpServletResponse response = (HttpServletResponse) servletResponse;
+            HttpSession session = request.getSession();
+
+            if (session.getAttribute(Constants.USERNAME) == null) {
+                //if the session does not exist
+                String token = request.getHeader(Constants.JWT_HEADER);
+                String[] tokenArray = token.split("\\.");
+                try {
+                    if (JwtTokenUtil.validateJWT(tokenArray, getJwtConfigList())) {
+                        //get the username from claims and assign it it session as attribute
+                        username = JwtTokenUtil.getUserNameFromJwt(tokenArray[1]);
+                        session.setAttribute(Constants.USERNAME, username);
+
+                        filterChain.doFilter(request, response);
+                    } else {
+                        throw new ScanManagerWebappException("Error while validating the JWT token");
```

Comment:
You already throws exceptions from the validateJWT method. Why throw from here again?